src/checkPassword.test.js

```diff
@@ -12,8 +12,62 @@ describe(`Function 'checkPassword':`, () => {
   });
 
   it(`should return 'true' for the valid password with 8 characters`, () => {
+    const result = checkPassword('Pass123!@');
 
+    expect(result).to.be.true;
+  });
+
+  it(`should return false for the password with 7 characters`, () => {
+    const result = checkPassword('Pass2!@');
+
+    expect(result).to.be.false;
+  });
+
+  it(`should return false for the password with 17 characters`, () => {
+    const result = checkPassword('Password123ijhda!');
+
+    expect(result).to.be.false;
+  });
+  it(`should return true for the password with 16 characters including at least 1 digit, 1 special character, 1 uppercase letter. `, () => {
+    const result = checkPassword('Passwordwertdc!1');
+
+    expect(result).to.be.false;
+  });
+
+  it('should return true for a valid password', () => {
+    const result = checkPassword('Password1!');
+
+    expect(result).to.be.true;
+  });
+
+  it('should return false for a password without special characters', () => {
+    const result = checkPassword('StrongPassword');
```

Comment:
this input should include digit

Reply:
not fixed, please make sure that all negative tests have only 1 deviation from the requirements